library/slice.lisp

```diff
@@ -33,9 +33,8 @@
   ;; Slice
   ;;
 
-  (repr :transparent)
-  (define-type (Slice :a)
-    (%Slice Lisp-Object))
+  (repr :native (cl:array cl:t (cl:*)))
```

Comment:
@stylewarning can you confirm that this is the correct type annotation for 1 dimensional potentially overlapping arrays of type `t`?

Reply:
yes this is correct, though you can simplify it to:

    (vector t)

which would be equivalent.

If you **know** it overlaps, you could write:

    (and (vector t) (not simple-vector))

To be clear, though, if the array/vector could possibly be specialized (like specialized single-float arrays), then you'll want

    (vector *)

or simply

     vector

This is different than `t`. 

`t` says "array of boxed pointers" (roughly)

`*` says "any kind of array/vector, possibly specialized, possibly not"